cchardet alternatives for python 3.11+

**Describe the bug**
For python 3.11  cChardet fails to build.  The error message is: `cchardet.cpp(196): fatal error C1083: Cannot open include file: 'longintrepr.h': No such file or directory.`

In addition, the cchardet doesn't seem to be currently maintained. 

This is a problem that will need to be resolved going forward.
`faust-streaming/cChardet` is a possible replacement.  It is a fork of cChardet.  
https://pypi.org/project/faust-cchardet/
https://github.com/faust-streaming/cChardet

**Notes**
Does not Support
[Anaconda](https://www.anaconda.com/)
[pyenv](https://github.com/pyenv/pyenv)

Does not yet install on MacOS
Cannot install on arm64 (Apple M1 Sillicon) #12


**To Reproduce**
Steps to reproduce the behavior:
- Add "3.11" to ci-test.yml on a dev branch and push to Github 
- GitHub->Actions->Python CI should fail for 3.11 image when trying to build cChardet.

**Expected behavior**
Python 3.11 builds/test on  GitHub->Actions->Python CI should should pass

**Software versions (please complete the following information):**
 - Python version: [3.11]
 - Python distribution: [GitHub]
 - lasio version: 0.30

**Additional context**
- I think `chardet` is a slower fall back option.  So there is a possibility that Lasio code/packaging can be arranged to use `chardet` on Python 3.11.

#### Additional Data

Tried install Lasio requirements in a Python 3.11 venv.  Here is the full cChardet install error:

```
 DEPRECATION: cchardet is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for cchardet ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for cchardet did not run successfully.
  │ exit code: 1
  ╰─> [25 lines of output]
      running install
      /Users/dcs/Wrk/Projects/geo-wrk/lasio/venvs/venv-3.11-lasio/lib/python3.11/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build/lib.macosx-13-arm64-cpython-311
      creating build/lib.macosx-13-arm64-cpython-311/cchardet
      copying src/cchardet/version.py -> build/lib.macosx-13-arm64-cpython-311/cchardet
      copying src/cchardet/__init__.py -> build/lib.macosx-13-arm64-cpython-311/cchardet
      running build_ext
      building 'cchardet._cchardet' extension
      creating build/temp.macosx-13-arm64-cpython-311
      creating build/temp.macosx-13-arm64-cpython-311/src
      creating build/temp.macosx-13-arm64-cpython-311/src/cchardet
      creating build/temp.macosx-13-arm64-cpython-311/src/ext
      creating build/temp.macosx-13-arm64-cpython-311/src/ext/uchardet
      creating build/temp.macosx-13-arm64-cpython-311/src/ext/uchardet/src
      creating build/temp.macosx-13-arm64-cpython-311/src/ext/uchardet/src/LangModels
      clang -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -Isrc/ext/uchardet/src -I/Users/dcs/Wrk/Projects/geo-wrk/lasio/venvs/venv-3.11-lasio/include -I/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c src/cchardet/_cchardet.cpp -o build/temp.macosx-13-arm64-cpython-311/src/cchardet/_cchardet.o
      src/cchardet/_cchardet.cpp:196:12: fatal error: 'longintrepr.h' file not found
        #include "longintrepr.h"
                 ^~~~~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> cchardet
```

--- a/.github/workflows/ci-tests.yml
+++ b/.github/workflows/ci-tests.yml
@@ -15,7 +15,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest]
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
 
     steps:
     - uses: actions/checkout@v3

--- a/README.md
+++ b/README.md
@@ -56,7 +56,7 @@ lasio with:
 $ pip install lasio
 ```
 
-To make sure you have everything, use this to ensure pandas, cchardet, and
+To make sure you have everything, use this to ensure pandas, chardet, and
 openpyxl are also installed:
 
 ```bash

--- a/docs/source/encodings.rst
+++ b/docs/source/encodings.rst
@@ -17,9 +17,8 @@ first one to raise no ``UnicodeDecodeError`` exceptions will be used.
 
 This may still result in an error, or incorrectly decoded characters.
 
-3. Install a package like `cChardet`_ (faster) or `chardet`_
-(slower) to automatically detect the character encoding. If these
-packages are installed then lasio will use them by default:
+3. Install optional package `chardet`_ to automatically detect the character
+   encoding. If `chardet`_ is installed then lasio will use it by default:
 
 .. code-block:: python
 
@@ -27,12 +26,10 @@ packages are installed then lasio will use them by default:
     >>> logging.basicConfig()
     >>> logging.getLogger().setLevel(logging.DEBUG)
     >>> las = lasio.read('encodings_utf8.las')
-    DEBUG:lasio.reader:get_encoding Using cchardet
-    DEBUG:lasio.reader:cchardet method detected encoding of UTF-8 at confidence 0.9900000095367432
-    INFO:lasio.reader:Opening encodings_utf8.las as UTF-8 and treating errors with "replace"
-    DEBUG:lasio.las:n_curves=8 ncols=8
-    DEBUG:lasio.las:set_data data.shape = (3, 8)
-    DEBUG:lasio.las:set_data self.data.shape = (3, 8)
+    DEBUG:lasio.reader:get_encoding Using chardet
+    DEBUG:lasio.reader:chardet method detected encoding of utf-8 at confidence 0.99
+    INFO:lasio.reader:Opening encodings_utf8.las as utf-8 and treating errors with "replace"
+    ...
 
 This may still result in an error, or incorrectly decoded characters.
 
@@ -48,5 +45,4 @@ using:
     ...     continue
 
 .. _open function from codecs: https://docs.python.org/2/library/codecs.html#codecs.open
-.. _cChardet: https://github.com/PyYoshi/cChardet
 .. _chardet: https://pypi.org/project/chardet

--- a/lasio/__init__.py
+++ b/lasio/__init__.py
@@ -124,9 +124,8 @@ def read(file_ref, **kwargs):
             <https://docs.python.org/3/library/codecs.html#codec-base-classes>`__
             of the standard library's :mod:`codecs` module for more information)
             (this is handled by :func:`lasio.reader.open_with_codecs`)
-        autodetect_encoding (str or bool): default True to use
-            `chardet <https://github.com/chardet/chardet>`__/`cchardet
-            <https://github.com/PyYoshi/cChardet>`__ to detect encoding.
+        autodetect_encoding (str or bool): default True to use `chardet
+            <https://github.com/chardet/chardet>`__ to detect encoding.
             Note if set to False several common encodings will be tried but
             chardet won't be used.
             (this is handled by :func:`lasio.reader.open_with_codecs`)

--- a/lasio/las.py
+++ b/lasio/las.py
@@ -98,9 +98,8 @@ class LASFile(object):
             <https://docs.python.org/3/library/codecs.html#codec-base-classes>`__
             of the standard library's :mod:`codecs` module for more information)
             (this is handled by :func:`lasio.reader.open_with_codecs`)
-        autodetect_encoding (str or bool): default True to use
-            `chardet <https://github.com/chardet/chardet>`__/`cchardet
-            <https://github.com/PyYoshi/cChardet>`__ to detect encoding.
+        autodetect_encoding (str or bool): default True to use `chardet
+            <https://github.com/chardet/chardet>`__ to detect encoding.
             Note if set to False several common encodings will be tried but
             chardet won't be used.
             (this is handled by :func:`lasio.reader.open_with_codecs`)
@@ -213,9 +212,8 @@ class LASFile(object):
                 <https://docs.python.org/3/library/codecs.html#codec-base-classes>`__
                 of the standard library's :mod:`codecs` module for more information)
                 (this is handled by :func:`lasio.reader.open_with_codecs`)
-            autodetect_encoding (str or bool): default True to use
-                `chardet <https://github.com/chardet/chardet>`__/`cchardet
-                <https://github.com/PyYoshi/cChardet>`__ to detect encoding.
+            autodetect_encoding (str or bool): default True to use `chardet
+                <https://github.com/chardet/chardet>`__ to detect encoding.
                 Note if set to False several common encodings will be tried but
                 chardet won't be used.
                 (this is handled by :func:`lasio.reader.open_with_codecs`)

--- a/lasio/reader.py
+++ b/lasio/reader.py
@@ -95,8 +95,8 @@ def check_for_path_obj(file_ref):
 def open_file(file_ref, **encoding_kwargs):
     """Open a file if necessary.
 
-    If ``autodetect_encoding=True`` then either ``cchardet`` or ``chardet``
-    needs to be installed, or else an ``ImportError`` will be raised.
+    If ``autodetect_encoding=True`` then  ``chardet`` needs to be installed, or
+    else an ``ImportError`` will be raised.
 
     Arguments:
         file_ref (file-like object, str): either a filename, an open file
@@ -162,8 +162,8 @@ def open_with_codecs(
             <https://docs.python.org/3/library/codecs.html#codec-base-classes>`__
             of the standard library's :mod:`codecs` module for more information)
         autodetect_encoding (str or bool): default True to use
-            `chardet <https://github.com/chardet/chardet>`__/`cchardet
-            <https://github.com/PyYoshi/cChardet>`__ to detect encoding.
+            `chardet <https://github.com/chardet/chardet>`__ to detect
+            encoding.
             Note if set to False several common encodings will be tried but
             chardet won't be used.
         autodetect_encoding_chars (int/None): number of chars to read from LAS
@@ -237,9 +237,9 @@ def get_encoding(auto, raw):
     Automatically detect character encoding.
 
     Arguments:
-        auto (str): auto-detection of character encoding - can be either
-            'chardet', 'cchardet', False, or True (the latter will pick the
-            fastest available option)
+        auto (str): auto-detection of character encoding - can be one of
+            'chardet', False, or True (the latter will pick the fastest
+            available option)
         raw (bytes): array of bytes to detect from
 
     Returns:
@@ -248,33 +248,21 @@ def get_encoding(auto, raw):
     """
     if auto is True:
         try:
-            import cchardet as chardet
+            import chardet
         except ImportError:
-            try:
-                import chardet
-            except ImportError:
-                logger.debug(
-                    "chardet or cchardet is recommended for automatic"
-                    " detection of character encodings. Instead trying some"
-                    " common encodings."
-                )
-                return None
-            else:
-                logger.debug("get_encoding Using chardet")
-                method = "chardet"
+            logger.debug(
+                "chardet is recommended for automatic detection of character"
+                "encodings. Instead trying some common encodings."
+            )
+            return None
         else:
-            logger.debug("get_encoding Using cchardet")
-            method = "cchardet"
+            logger.debug("get_encoding Using chardet")
+            method = "chardet"
     elif auto.lower() == "chardet":
         import chardet
 
         logger.debug("get_encoding Using chardet")
         method = "chardet"
-    elif auto.lower() == "cchardet":
-        import cchardet as chardet
-
-        logger.debug("get_encoding Using cchardet")
-        method = "cchardet"
     result = chardet.detect(raw)
     logger.debug(
         "{} method detected encoding of {} at confidence {}".format(

--- a/notebooks/set logging level for lasio.ipynb
+++ b/notebooks/set logging level for lasio.ipynb
@@ -60,14 +60,14 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 6,
+   "execution_count": 4,
    "metadata": {},
    "outputs": [
     {
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "INFO:lasio.reader:Opening ../tests/examples/sample.las as ASCII and treating errors with \"replace\"\n"
+      "INFO:lasio.reader:Opening ../tests/examples/sample.las as ascii and treating errors with \"replace\"\n"
      ]
     }
    ],
@@ -87,7 +87,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 7,
+   "execution_count": 5,
    "metadata": {},
    "outputs": [
     {
@@ -97,28 +97,16 @@
       "DEBUG:lasio.las:Reading ../tests/examples/sample.las...\n",
       "DEBUG:lasio.las:Ignore header lines beginning with ('#',)\n",
       "DEBUG:lasio.las:Ignore data lines beginning with #\n",
-      "DEBUG:lasio.reader:get_encoding Using cchardet\n",
-      "DEBUG:lasio.reader:cchardet method detected encoding of ASCII at confidence 1.0\n",
-      "INFO:lasio.reader:Opening ../tests/examples/sample.las as ASCII and treating errors with \"replace\"\n",
+      "DEBUG:lasio.reader:get_encoding Using chardet\n",
+      "DEBUG:lasio.reader:chardet method detected encoding of ascii at confidence 1.0\n",
+      "INFO:lasio.reader:Opening ../tests/examples/sample.las as ascii and treating errors with \"replace\"\n",
       "DEBUG:lasio.las:Fetching substitutions for read_policy default and null policy strict\n",
-      "DEBUG:lasio.reader:using read policy of \"default\"\n",
-      "DEBUG:lasio.reader:adding substitution comma-decimal-mark\n",
-      "DEBUG:lasio.reader:adding substitution run-on(-)\n",
-      "DEBUG:lasio.reader:adding substitution run-on(.)\n",
-      "DEBUG:lasio.reader:adding substitution run-on(NaN.)\n",
-      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('(\\\\d),(\\\\d)') substr=\"\\1.\\2\"\n",
-      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('(\\\\d)-(\\\\d)') substr=\"\\1 -\\2\"\n",
-      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('-?\\\\d*\\\\.\\\\d*\\\\.\\\\d*|NaN[\\\\.-]\\\\d+') substr=\" NaN NaN \"\n",
-      "DEBUG:lasio.reader:using null policy of \"strict\"\n",
-      "DEBUG:lasio.reader:adding substitution NULL\n",
-      "DEBUG:lasio.reader:located substitution for LAS.version.NULL as True\n",
-      "DEBUG:lasio.reader:added numerical substitution: None\n",
       "DEBUG:lasio.las:Found 6 sections\n",
       "DEBUG:lasio.las:Parsing Header items section at lines 1-3 (0 bytes) ~VERSION INFORMATION\n",
       "DEBUG:lasio.reader:Line 1: Section title parsed as '~VERSION INFORMATION'\n",
       "DEBUG:lasio.reader:Line 2: parsed as HeaderItem(mnemonic=\"VERS\", unit=\"\", value=\"1.2\", descr=\"CWLS LOG ASCII STANDA\")\n",
       "DEBUG:lasio.reader:Line 3: parsed as HeaderItem(mnemonic=\"WRAP\", unit=\"\", value=\"NO\", descr=\"ONE LINE PER DEPTH STE\")\n",
-      "DEBUG:lasio.las:Parsing Header items section at lines 4-18 (146 bytes) ~WELL INFORMATION BLOCK\n",
+      "DEBUG:lasio.las:Parsing Header items section at lines 4-18 (143 bytes) ~WELL INFORMATION BLOCK\n",
       "DEBUG:lasio.reader:Line 4: Section title parsed as '~WELL INFORMATION BLOCK'\n",
       "DEBUG:lasio.reader:Line 5: treating as a comment and ignoring: '#MNEM.UNIT       DATA TYPE    INFORMATION'\n",
       "DEBUG:lasio.reader:Line 6: treating as a comment and ignoring: '#---------    -------------   ------------------------------'\n",
@@ -134,7 +122,7 @@
       "DEBUG:lasio.reader:Line 16: parsed as HeaderItem(mnemonic=\"SRVC\", unit=\"\", value=\"ANY LOGGING COMPANY LTD.\", descr=\"\")\n",
       "DEBUG:lasio.reader:Line 17: parsed as HeaderItem(mnemonic=\"DATE\", unit=\"\", value=\"25-DEC-1988\", descr=\"LOG DATE\")\n",
       "DEBUG:lasio.reader:Line 18: parsed as HeaderItem(mnemonic=\"UWI\", unit=\"\", value=\"100091604920W300\", descr=\"UNIQUE WE\")\n",
-      "DEBUG:lasio.las:Parsing Header items section at lines 19-29 (773 bytes) ~CURVE INFORMATION\n",
+      "DEBUG:lasio.las:Parsing Header items section at lines 19-29 (755 bytes) ~CURVE INFORMATION\n",
       "DEBUG:lasio.reader:Line 19: Section title parsed as '~CURVE INFORMATION'\n",
       "DEBUG:lasio.reader:Line 20: treating as a comment and ignoring: '#MNEM.UNIT      API CODE      CURVE DESCRIPTION'\n",
       "DEBUG:lasio.reader:Line 21: treating as a comment and ignoring: '#---------    -------------   ------------------------------'\n",
@@ -146,7 +134,7 @@
       "DEBUG:lasio.reader:Line 27: parsed as CurveItem(mnemonic=\"SFLA\", unit=\"OHMM\", value=\"\", descr=\"6  SHALLOW RESISTIVITY\", original_mnemonic=\"SFLA\", data.shape=(0,))\n",
       "DEBUG:lasio.reader:Line 28: parsed as CurveItem(mnemonic=\"ILM\", unit=\"OHMM\", value=\"\", descr=\"7  MEDIUM RESISTIVITY\", original_mnemonic=\"ILM\", data.shape=(0,))\n",
       "DEBUG:lasio.reader:Line 29: parsed as CurveItem(mnemonic=\"ILD\", unit=\"OHMM\", value=\"\", descr=\"8  DEEP RESISTIVITY\", original_mnemonic=\"ILD\", data.shape=(0,))\n",
-      "DEBUG:lasio.las:Parsing Header items section at lines 30-39 (1313 bytes) ~PARAMETER INFORMATION\n",
+      "DEBUG:lasio.las:Parsing Header items section at lines 30-39 (1284 bytes) ~PARAMETER INFORMATION\n",
       "DEBUG:lasio.reader:Line 30: Section title parsed as '~PARAMETER INFORMATION'\n",
       "DEBUG:lasio.reader:Line 31: treating as a comment and ignoring: '#MNEM.UNIT        VALUE       DESCRIPTION'\n",
       "DEBUG:lasio.reader:Line 32: treating as a comment and ignoring: '#---------    -------------   ------------------------------'\n",
@@ -157,9 +145,20 @@
       "DEBUG:lasio.reader:Line 37: parsed as HeaderItem(mnemonic=\"MDEN\", unit=\"\", value=\"2710.0\", descr=\"LOGGING MATRIX DEN\")\n",
       "DEBUG:lasio.reader:Line 38: parsed as HeaderItem(mnemonic=\"RMF\", unit=\"OHMM\", value=\"0.216\", descr=\"MUD FILTRATE RES\")\n",
       "DEBUG:lasio.reader:Line 39: parsed as HeaderItem(mnemonic=\"DFD\", unit=\"K/M3\", value=\"1525.0\", descr=\"DRILL FLUID DEN\")\n",
-      "DEBUG:lasio.las:Parsing Header (other) section at lines 40-42 (1811 bytes) ~Other\n",
-      "DEBUG:lasio.las:Parsing Data section at lines 43-47 (1947 bytes) ~A  DEPTH     DT       RHOB     NPHI     SFLU     SFLA      ILM      ILD\n",
+      "DEBUG:lasio.las:Parsing Header (other) section at lines 40-42 (1772 bytes) ~Other\n",
+      "DEBUG:lasio.las:Parsing Data section at lines 43-47 (1905 bytes) ~A  DEPTH     DT       RHOB     NPHI     SFLU     SFLA      ILM      ILD\n",
       "DEBUG:lasio.las:Storing reference and returning later...\n",
+      "DEBUG:lasio.reader:using read policy of \"default\"\n",
+      "DEBUG:lasio.reader:adding substitution comma-decimal-mark\n",
+      "DEBUG:lasio.reader:adding substitution run-on(-)\n",
+      "DEBUG:lasio.reader:adding substitution run-on(.)\n",
+      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('(\\\\d),(\\\\d)') substr=\"\\1.\\2\"\n",
+      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('(\\\\d)-(\\\\d)') substr=\"\\1 -\\2\"\n",
+      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('-?\\\\d*\\\\.\\\\d*\\\\.\\\\d*|NaN[\\\\.-]\\\\d+') substr=\" NaN NaN \"\n",
+      "DEBUG:lasio.reader:using null policy of \"strict\"\n",
+      "DEBUG:lasio.reader:adding substitution NULL\n",
+      "DEBUG:lasio.reader:located substitution for LAS.version.NULL as True\n",
+      "DEBUG:lasio.reader:added numerical substitution: None\n",
       "DEBUG:lasio.las:Reading data section ~A  DEPTH     DT       RHOB     NPHI     SFLU     SFLA      ILM      ILD\n",
       "DEBUG:lasio.reader:Consistently found 8 columns\n",
       "DEBUG:lasio.las:Assigning data [1670.    1669.875 1669.75 ] to curve #0\n",
@@ -205,7 +204,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 9,
+   "execution_count": 6,
    "metadata": {},
    "outputs": [
     {
@@ -215,28 +214,16 @@
       "DEBUG:lasio.las:Reading ../tests/examples/sample.las...\n",
       "DEBUG:lasio.las:Ignore header lines beginning with ('#',)\n",
       "DEBUG:lasio.las:Ignore data lines beginning with #\n",
-      "DEBUG:lasio.reader:get_encoding Using cchardet\n",
-      "DEBUG:lasio.reader:cchardet method detected encoding of ASCII at confidence 1.0\n",
-      "INFO:lasio.reader:Opening ../tests/examples/sample.las as ASCII and treating errors with \"replace\"\n",
+      "DEBUG:lasio.reader:get_encoding Using chardet\n",
+      "DEBUG:lasio.reader:chardet method detected encoding of ascii at confidence 1.0\n",
+      "INFO:lasio.reader:Opening ../tests/examples/sample.las as ascii and treating errors with \"replace\"\n",
       "DEBUG:lasio.las:Fetching substitutions for read_policy default and null policy strict\n",
-      "DEBUG:lasio.reader:using read policy of \"default\"\n",
-      "DEBUG:lasio.reader:adding substitution comma-decimal-mark\n",
-      "DEBUG:lasio.reader:adding substitution run-on(-)\n",
-      "DEBUG:lasio.reader:adding substitution run-on(.)\n",
-      "DEBUG:lasio.reader:adding substitution run-on(NaN.)\n",
-      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('(\\\\d),(\\\\d)') substr=\"\\1.\\2\"\n",
-      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('(\\\\d)-(\\\\d)') substr=\"\\1 -\\2\"\n",
-      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('-?\\\\d*\\\\.\\\\d*\\\\.\\\\d*|NaN[\\\\.-]\\\\d+') substr=\" NaN NaN \"\n",
-      "DEBUG:lasio.reader:using null policy of \"strict\"\n",
-      "DEBUG:lasio.reader:adding substitution NULL\n",
-      "DEBUG:lasio.reader:located substitution for LAS.version.NULL as True\n",
-      "DEBUG:lasio.reader:added numerical substitution: None\n",
       "DEBUG:lasio.las:Found 6 sections\n",
       "DEBUG:lasio.las:Parsing Header items section at lines 1-3 (0 bytes) ~VERSION INFORMATION\n",
       "DEBUG:lasio.reader:Line 1: Section title parsed as '~VERSION INFORMATION'\n",
       "DEBUG:lasio.reader:Line 2: parsed as HeaderItem(mnemonic=\"VERS\", unit=\"\", value=\"1.2\", descr=\"CWLS LOG ASCII STANDA\")\n",
       "DEBUG:lasio.reader:Line 3: parsed as HeaderItem(mnemonic=\"WRAP\", unit=\"\", value=\"NO\", descr=\"ONE LINE PER DEPTH STE\")\n",
-      "DEBUG:lasio.las:Parsing Header items section at lines 4-18 (146 bytes) ~WELL INFORMATION BLOCK\n",
+      "DEBUG:lasio.las:Parsing Header items section at lines 4-18 (143 bytes) ~WELL INFORMATION BLOCK\n",
       "DEBUG:lasio.reader:Line 4: Section title parsed as '~WELL INFORMATION BLOCK'\n",
       "DEBUG:lasio.reader:Line 5: treating as a comment and ignoring: '#MNEM.UNIT       DATA TYPE    INFORMATION'\n",
       "DEBUG:lasio.reader:Line 6: treating as a comment and ignoring: '#---------    -------------   ------------------------------'\n",
@@ -252,7 +239,7 @@
       "DEBUG:lasio.reader:Line 16: parsed as HeaderItem(mnemonic=\"SRVC\", unit=\"\", value=\"ANY LOGGING COMPANY LTD.\", descr=\"\")\n",
       "DEBUG:lasio.reader:Line 17: parsed as HeaderItem(mnemonic=\"DATE\", unit=\"\", value=\"25-DEC-1988\", descr=\"LOG DATE\")\n",
       "DEBUG:lasio.reader:Line 18: parsed as HeaderItem(mnemonic=\"UWI\", unit=\"\", value=\"100091604920W300\", descr=\"UNIQUE WE\")\n",
-      "DEBUG:lasio.las:Parsing Header items section at lines 19-29 (773 bytes) ~CURVE INFORMATION\n",
+      "DEBUG:lasio.las:Parsing Header items section at lines 19-29 (755 bytes) ~CURVE INFORMATION\n",
       "DEBUG:lasio.reader:Line 19: Section title parsed as '~CURVE INFORMATION'\n",
       "DEBUG:lasio.reader:Line 20: treating as a comment and ignoring: '#MNEM.UNIT      API CODE      CURVE DESCRIPTION'\n",
       "DEBUG:lasio.reader:Line 21: treating as a comment and ignoring: '#---------    -------------   ------------------------------'\n",
@@ -264,7 +251,7 @@
       "DEBUG:lasio.reader:Line 27: parsed as CurveItem(mnemonic=\"SFLA\", unit=\"OHMM\", value=\"\", descr=\"6  SHALLOW RESISTIVITY\", original_mnemonic=\"SFLA\", data.shape=(0,))\n",
       "DEBUG:lasio.reader:Line 28: parsed as CurveItem(mnemonic=\"ILM\", unit=\"OHMM\", value=\"\", descr=\"7  MEDIUM RESISTIVITY\", original_mnemonic=\"ILM\", data.shape=(0,))\n",
       "DEBUG:lasio.reader:Line 29: parsed as CurveItem(mnemonic=\"ILD\", unit=\"OHMM\", value=\"\", descr=\"8  DEEP RESISTIVITY\", original_mnemonic=\"ILD\", data.shape=(0,))\n",
-      "DEBUG:lasio.las:Parsing Header items section at lines 30-39 (1313 bytes) ~PARAMETER INFORMATION\n",
+      "DEBUG:lasio.las:Parsing Header items section at lines 30-39 (1284 bytes) ~PARAMETER INFORMATION\n",
       "DEBUG:lasio.reader:Line 30: Section title parsed as '~PARAMETER INFORMATION'\n",
       "DEBUG:lasio.reader:Line 31: treating as a comment and ignoring: '#MNEM.UNIT        VALUE       DESCRIPTION'\n",
       "DEBUG:lasio.reader:Line 32: treating as a comment and ignoring: '#---------    -------------   ------------------------------'\n",
@@ -275,9 +262,20 @@
       "DEBUG:lasio.reader:Line 37: parsed as HeaderItem(mnemonic=\"MDEN\", unit=\"\", value=\"2710.0\", descr=\"LOGGING MATRIX DEN\")\n",
       "DEBUG:lasio.reader:Line 38: parsed as HeaderItem(mnemonic=\"RMF\", unit=\"OHMM\", value=\"0.216\", descr=\"MUD FILTRATE RES\")\n",
       "DEBUG:lasio.reader:Line 39: parsed as HeaderItem(mnemonic=\"DFD\", unit=\"K/M3\", value=\"1525.0\", descr=\"DRILL FLUID DEN\")\n",
-      "DEBUG:lasio.las:Parsing Header (other) section at lines 40-42 (1811 bytes) ~Other\n",
-      "DEBUG:lasio.las:Parsing Data section at lines 43-47 (1947 bytes) ~A  DEPTH     DT       RHOB     NPHI     SFLU     SFLA      ILM      ILD\n",
+      "DEBUG:lasio.las:Parsing Header (other) section at lines 40-42 (1772 bytes) ~Other\n",
+      "DEBUG:lasio.las:Parsing Data section at lines 43-47 (1905 bytes) ~A  DEPTH     DT       RHOB     NPHI     SFLU     SFLA      ILM      ILD\n",
       "DEBUG:lasio.las:Storing reference and returning later...\n",
+      "DEBUG:lasio.reader:using read policy of \"default\"\n",
+      "DEBUG:lasio.reader:adding substitution comma-decimal-mark\n",
+      "DEBUG:lasio.reader:adding substitution run-on(-)\n",
+      "DEBUG:lasio.reader:adding substitution run-on(.)\n",
+      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('(\\\\d),(\\\\d)') substr=\"\\1.\\2\"\n",
+      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('(\\\\d)-(\\\\d)') substr=\"\\1 -\\2\"\n",
+      "DEBUG:lasio.reader:added regexp substitution: pattern=re.compile('-?\\\\d*\\\\.\\\\d*\\\\.\\\\d*|NaN[\\\\.-]\\\\d+') substr=\" NaN NaN \"\n",
+      "DEBUG:lasio.reader:using null policy of \"strict\"\n",
+      "DEBUG:lasio.reader:adding substitution NULL\n",
+      "DEBUG:lasio.reader:located substitution for LAS.version.NULL as True\n",
+      "DEBUG:lasio.reader:added numerical substitution: None\n",
       "DEBUG:lasio.las:Reading data section ~A  DEPTH     DT       RHOB     NPHI     SFLU     SFLA      ILM      ILD\n",
       "TRACE_LASIO:lasio.reader:Line 44: 8 items counted in '1670.000   123.450 2550.000    0.450  123.450  123.450  110.200  105.600'\n",
       "TRACE_LASIO:lasio.reader:Line 45: 8 items counted in '1669.875   123.450 2550.000    0.450  123.450  123.450  110.200  105.600'\n",
@@ -316,7 +314,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 10,
+   "execution_count": 7,
    "metadata": {},
    "outputs": [],
    "source": [
@@ -336,7 +334,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 11,
+   "execution_count": 8,
    "metadata": {
     "scrolled": true
    },
@@ -347,7 +345,7 @@
        "HeaderItem(mnemonic=\"SRVC\", unit=\"\", value=\"ANY LOGGING COMPANY LTD.\", descr=\"\")"
       ]
      },
-     "execution_count": 11,
+     "execution_count": 8,
      "metadata": {},
      "output_type": "execute_result"
     }
@@ -355,20 +353,13 @@
    "source": [
     "l.header['Well'].SRVC"
    ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": []
   }
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python [conda env:lasiodev]",
+   "display_name": "Python 3 (ipykernel)",
    "language": "python",
-   "name": "conda-env-lasiodev-py"
+   "name": "python3"
   },
   "language_info": {
    "codemirror_mode": {
@@ -380,7 +371,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.9.6"
+   "version": "3.10.9"
   }
  },
  "nbformat": 4,

--- a/setup.py
+++ b/setup.py
@@ -3,7 +3,7 @@
 from setuptools import setup
 import os
 
-EXTRA_REQS = ("pandas", "cchardet", "openpyxl")
+EXTRA_REQS = ("pandas", "chardet", "openpyxl")
 TEST_REQS = (
     "pytest>=3.6",
     "pytest-cov",

--- a/tests/test_encoding.py
+++ b/tests/test_encoding.py
@@ -16,20 +16,20 @@ stegfn = lambda vers, fn: os.path.join(os.path.dirname(__file__), "examples", ve
 
 
 def test_encoding_attr():
-    las = read(egfn("encodings_utf8.las"), autodetect_encoding="cchardet")
-    assert las.encoding == "UTF-8"
+    las = read(egfn("encodings_utf8.las"), autodetect_encoding="chardet")
+    assert las.encoding.upper() == "UTF-8"
 
 
-def test_utf8_cchardet():
-    las = read(egfn("encodings_utf8.las"), autodetect_encoding="cchardet")
+def test_utf8_chardet():
+    las = read(egfn("encodings_utf8.las"), autodetect_encoding="chardet")
 
 
-def test_utf8wbom_cchardet():
-    las = read(egfn("encodings_utf8wbom.las"), autodetect_encoding="cchardet")
+def test_utf8wbom_chardet():
+    las = read(egfn("encodings_utf8wbom.las"), autodetect_encoding="chardet")
 
 
-def test_utf16lebom_cchardet():
-    las = read(egfn("encodings_utf16lebom.las"), autodetect_encoding="cchardet")
+def test_utf16lebom_chardet():
+    las = read(egfn("encodings_utf16lebom.las"), autodetect_encoding="chardet")
 
 
 def test_utf16le_specified_ok():
@@ -37,21 +37,21 @@ def test_utf16le_specified_ok():
 
 
 @pytest.mark.skip(reason="this is not behaving properly see PR #326")
-def test_utf16le_cchardet_fails():
+def test_utf16le_chardet_fails():
     with pytest.raises(Exception):
-        las = read(egfn("encodings_utf16le.las"), autodetect_encoding="cchardet")
+        las = read(egfn("encodings_utf16le.las"), autodetect_encoding="chardet")
 
 
-def test_utf16bebom_cchardet():
-    las = read(egfn("encodings_utf16bebom.las"), autodetect_encoding="cchardet")
+def test_utf16bebom_chardet():
+    las = read(egfn("encodings_utf16bebom.las"), autodetect_encoding="chardet")
 
 
-def test_iso88591_cchardet():
-    las = read(egfn("encodings_iso88591.las"), autodetect_encoding="cchardet")
+def test_iso88591_chardet():
+    las = read(egfn("encodings_iso88591.las"), autodetect_encoding="chardet")
 
 
-def test_cp1252_cchardet():
-    las = read(egfn("encodings_cp1252.las"), autodetect_encoding="cchardet")
+def test_cp1252_chardet():
+    las = read(egfn("encodings_cp1252.las"), autodetect_encoding="chardet")
 
 
 """
@@ -59,16 +59,16 @@ Verify encodings for pathlib.Path objects
 """
 
 
-def test_pathlib_utf8_cchardet():
-    las = read(Path(egfn("encodings_utf8.las")), autodetect_encoding="cchardet")
+def test_pathlib_utf8_chardet():
+    las = read(Path(egfn("encodings_utf8.las")), autodetect_encoding="chardet")
 
 
-def test_pathlib_utf8wbom_cchardet():
-    las = read(Path(egfn("encodings_utf8wbom.las")), autodetect_encoding="cchardet")
+def test_pathlib_utf8wbom_chardet():
+    las = read(Path(egfn("encodings_utf8wbom.las")), autodetect_encoding="chardet")
 
 
-def test_pathlib_utf16lebom_cchardet():
-    las = read(Path(egfn("encodings_utf16lebom.las")), autodetect_encoding="cchardet")
+def test_pathlib_utf16lebom_chardet():
+    las = read(Path(egfn("encodings_utf16lebom.las")), autodetect_encoding="chardet")
 
 
 def test_pathlib_utf16le_specified_ok():
@@ -76,21 +76,21 @@ def test_pathlib_utf16le_specified_ok():
 
 
 @pytest.mark.skip(reason="this is not behaving properly see PR #326")
-def test_pathlib_utf16le_cchardet_fails():
+def test_pathlib_utf16le_chardet_fails():
     with pytest.raises(Exception):
-        las = read(Path(egfn("encodings_utf16le.las")), autodetect_encoding="cchardet")
+        las = read(Path(egfn("encodings_utf16le.las")), autodetect_encoding="chardet")
 
 
-def test_pathlib_utf16bebom_cchardet():
-    las = read(Path(egfn("encodings_utf16bebom.las")), autodetect_encoding="cchardet")
+def test_pathlib_utf16bebom_chardet():
+    las = read(Path(egfn("encodings_utf16bebom.las")), autodetect_encoding="chardet")
 
 
-def test_pathlib_iso88591_cchardet():
-    las = read(Path(egfn("encodings_iso88591.las")), autodetect_encoding="cchardet")
+def test_pathlib_iso88591_chardet():
+    las = read(Path(egfn("encodings_iso88591.las")), autodetect_encoding="chardet")
 
 
-def test_pathlib_cp1252_cchardet():
-    las = read(Path(egfn("encodings_cp1252.las")), autodetect_encoding="cchardet")
+def test_pathlib_cp1252_chardet():
+    las = read(Path(egfn("encodings_cp1252.las")), autodetect_encoding="chardet")
 
 
 def test_adhoc_test_encoding():
@@ -108,4 +108,4 @@ def test_open_with_codecs_no_autodetect():
 def test_open_with_codecs_no_autodetect_chars():
     filename = stegfn("1.2", "sample.las")
     obj, encoding = reader.open_with_codecs(filename, autodetect_encoding_chars=0)
-    assert encoding == "ASCII"
+    assert encoding.upper() == "ASCII"

